dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBStreamHandler.java

```diff
@@ -0,0 +1,121 @@
+package no.unit.nva.dynamodb;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.DynamodbEvent;
+import com.amazonaws.services.lambda.runtime.events.models.dynamodb.AttributeValue;
+import elasticsearch.ElasticSearchRestClient;
+import nva.commons.utils.Environment;
+import nva.commons.utils.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+
+public class DynamoDBStreamHandler implements RequestHandler<DynamodbEvent, String> {
+
+    private static final Logger logger = LoggerFactory.getLogger(DynamoDBStreamHandler.class);
+
+    public static final String IDENTIFIER = "identifier";
+    public static final String DATE_YEAR = "entityDescription.date.year";
+    public static final String DESCRIPTION_MAIN_TITLE = "entityDescription.mainTitle";
+    public static final String CONTRIBUTORS_IDENTITY_NAME = "entityDescription.contributors.identity.name";
+    public static final String PUBLICATION_TYPE = "publicationInstance.type";
+    public static final String YEAR = "year";
+    public static final String TITLE = "title";
+    public static final String NAME = "author";
+    public static final String TYPE = "publicationType";
+    private final ElasticSearchRestClient elasticSearchClient;
+
+    /**
+     * Default constructor for DynamoDBStreamHandler.
+     */
+    @JacocoGenerated
+    public DynamoDBStreamHandler() {
+        this(new ElasticSearchRestClient(HttpClient.newHttpClient(), new Environment()));
+    }
+
+    /**
+     * Constructor for DynamoDBStreamHandler for testing.
+     * @param elasticSearchRestClient elasticSearchRestClient to be injected for testing
+     */
+    @JacocoGenerated
+    public DynamoDBStreamHandler(ElasticSearchRestClient elasticSearchRestClient) {
+        this.elasticSearchClient = elasticSearchRestClient;
+    }
+
+
+
+    @Override
+    public String handleRequest(DynamodbEvent event, Context context) {
+        for (DynamodbEvent.DynamodbStreamRecord streamRecord : event.getRecords()) {
+            switch (streamRecord.getEventName()) {
+                case "INSERT":
+                case "MODIFY":
+                    try {
```

Comment:
this block should be a method.

Reply:
Errorhandling changed to make the wole event fail